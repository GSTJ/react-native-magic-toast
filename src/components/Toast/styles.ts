import { StyleSheet, TextStyle, ViewStyle } from 'react-native';

interface MagicToastStyles {
  container: ViewStyle;
  message: TextStyle;
}

export const styles = StyleSheet.create<MagicToastStyles>({
  container: {
    backgroundColor: '#191919',
    padding: 25,
    flexDirection: 'row',
    alignItems: 'center',
    gap: 15,
  },
  message: {
    color: 'white',
    fontWeight: 'bold',
  },
});
