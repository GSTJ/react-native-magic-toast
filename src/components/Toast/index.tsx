import React, { useEffect } from 'react';
import { View, ViewProps, TextProps, Text } from 'react-native';
import { magicModal } from 'react-native-magic-modal';
import { useSafeAreaInsets } from 'react-native-safe-area-context';
import { styles } from './styles';

export const TOAST_TEST_ID = 'magic-toast';

interface ContainerProps extends ViewProps {
  duration?: number;
}

/**
 * The container of the toast. It is responsible for hiding the toast after a
 * certain amount of time.
 * @param props.duration The duration of the toast.
 * @example
 *  <Toast.Container duration={3000}>
 *    <Toast.Message>My message</Toast.Message>
 *  </Toast.Container>
 */
const Container: React.FC<ContainerProps> = ({ duration = 3000, ...props }) => {
  const { top } = useSafeAreaInsets();

  useEffect(() => {
    const timeout = setTimeout(magicModal.hide, duration);
    return () => clearTimeout(timeout);
  }, [duration]);

  return (
    <View
      testID={TOAST_TEST_ID}
      {...props}
      style={[styles.container, { paddingTop: 25 + top }, props.style]}
    />
  );
};

/**
 * A default Toast message component to help you get started.
 * @example
 *  <Toast.Container duration={3000}>
 *    <Toast.Message>My message</Toast.Message>
 *  </Toast.Container>
 */
const Message: React.FC<TextProps> = (props) => (
  <Text {...props} style={[styles.message, props.style]} />
);

export const Toast = {
  Container,
  Message,
};
