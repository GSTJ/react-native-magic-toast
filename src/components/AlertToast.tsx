import React from 'react';
import { StatusBar } from 'react-native';
import { AlertIcon } from '../assets/AlertIcon';
import { Toast } from './Toast';

import type { MagicToastProps } from 'src/@types/MagicToastProps';

export const AlertToast: React.FC<MagicToastProps> = ({
  message,
  duration,
}) => {
  return (
    <Toast.Container duration={duration}>
      <StatusBar barStyle="light-content" />
      <AlertIcon fill="white" width={20} height={20} />
      <Toast.Message>{message}</Toast.Message>
    </Toast.Container>
  );
};
