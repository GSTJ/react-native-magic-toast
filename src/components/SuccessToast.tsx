import React from 'react';
import { StatusBar, StyleSheet } from 'react-native';
import { SuccessIcon } from '../assets/SuccessIcon';
import { Toast } from './Toast';

import type { MagicToastProps } from 'src/@types/MagicToastProps';

export const SuccessToast: React.FC<MagicToastProps> = ({
  message,
  duration,
}) => {
  return (
    <Toast.Container style={styles.container} duration={duration}>
      <StatusBar barStyle="light-content" />
      <SuccessIcon fill="white" width={25} height={25} />
      <Toast.Message>{message}</Toast.Message>
    </Toast.Container>
  );
};

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#00af98',
    gap: 10,
  },
});
