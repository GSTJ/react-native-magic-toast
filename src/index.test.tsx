import React from 'react';
import { render, act } from '@testing-library/react-native';
import { MagicModalPortal } from 'react-native-magic-modal';
import { magicToast } from './index';
import { SafeAreaProvider } from 'react-native-safe-area-context';
import { TOAST_TEST_ID } from './components/Toast';

describe('MagicToast', () => {
  it('renders alerts correctly', async () => {
    const component = render(
      <SafeAreaProvider>
        <MagicModalPortal />
      </SafeAreaProvider>
    );

    expect(component.queryByTestId(TOAST_TEST_ID)).toBeFalsy();

    act(() => {
      magicToast.alert('Taveira');
    });

    expect(component).toMatchSnapshot();
    expect(component.queryByTestId(TOAST_TEST_ID)).toBeTruthy();
  });

  it('renders success correctly', async () => {
    const component = render(
      <SafeAreaProvider>
        <MagicModalPortal />
      </SafeAreaProvider>
    );

    expect(component.queryByTestId(TOAST_TEST_ID)).toBeFalsy();

    act(() => {
      magicToast.success('Taveira');
    });

    expect(component).toMatchSnapshot();
    expect(component.queryByTestId(TOAST_TEST_ID)).toBeTruthy();
  });
});
