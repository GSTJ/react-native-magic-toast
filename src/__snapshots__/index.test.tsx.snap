// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MagicToast renders alerts correctly 1`] = `
<View
  pointerEvents="box-none"
  style={
    [
      {
        "backgroundColor": "black",
        "bottom": 0,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      },
      {
        "backgroundColor": "transparent",
        "opacity": 1,
        "zIndex": 2,
      },
    ]
  }
>
  <View
    collapsable={false}
    deviceHeight={1334}
    deviceWidth={750}
    hideModalContentWhileAnimating={false}
    onBackdropPress={[Function]}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    onSwipeComplete={[Function]}
    panResponderThreshold={4}
    pointerEvents="box-none"
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    statusBarTranslucent={true}
    style={
      {
        "flex": 1,
        "justifyContent": "flex-start",
        "left": 0,
        "margin": 0,
        "opacity": 0,
        "top": 0,
        "transform": [
          {
            "translateY": -100,
          },
        ],
      }
    }
    supportedOrientations={
      [
        "portrait",
        "landscape",
      ]
    }
    swipeDirection="up"
    swipeThreshold={100}
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "backgroundColor": "#191919",
            "flexDirection": "row",
            "gap": 15,
            "padding": 25,
          },
          {
            "paddingTop": 25,
          },
          undefined,
        ]
      }
      testID="magic-toast"
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight={20}
        bbWidth={20}
        fill="white"
        focusable={false}
        height={20}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "flex": 0,
              "height": 20,
              "width": 20,
            },
          ]
        }
        vbHeight={512}
        vbWidth={512}
        width={20}
        x="0px"
        y="0px"
      >
        <RNSVGGroup
          fill={4294967295}
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="M501.362 383.95L320.497 51.474c-29.059-48.921-99.896-48.986-128.994 0L10.647 383.95c-29.706 49.989 6.259 113.291 64.482 113.291h361.736c58.174 0 94.203-63.251 64.497-113.291zM256 437.241c-16.538 0-30-13.462-30-30s13.462-30 30-30 30 13.462 30 30-13.462 30-30 30zm30-120c0 16.538-13.462 30-30 30s-30-13.462-30-30v-150c0-16.538 13.462-30 30-30s30 13.462 30 30v150z"
          />
        </RNSVGGroup>
      </RNSVGSvgView>
      <Text
        style={
          [
            {
              "color": "white",
              "fontWeight": "bold",
            },
            undefined,
          ]
        }
      >
        Taveira
      </Text>
    </View>
  </View>
</View>
`;

exports[`MagicToast renders success correctly 1`] = `
<View
  pointerEvents="box-none"
  style={
    [
      {
        "backgroundColor": "black",
        "bottom": 0,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      },
      {
        "backgroundColor": "transparent",
        "opacity": 1,
        "zIndex": 2,
      },
    ]
  }
>
  <View
    collapsable={false}
    deviceHeight={1334}
    deviceWidth={750}
    hideModalContentWhileAnimating={false}
    onBackdropPress={[Function]}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    onSwipeComplete={[Function]}
    panResponderThreshold={4}
    pointerEvents="box-none"
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    statusBarTranslucent={true}
    style={
      {
        "flex": 1,
        "justifyContent": "flex-start",
        "left": 0,
        "margin": 0,
        "opacity": 0,
        "top": 0,
        "transform": [
          {
            "translateY": -100,
          },
        ],
      }
    }
    supportedOrientations={
      [
        "portrait",
        "landscape",
      ]
    }
    swipeDirection="up"
    swipeThreshold={100}
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "backgroundColor": "#191919",
            "flexDirection": "row",
            "gap": 15,
            "padding": 25,
          },
          {
            "paddingTop": 25,
          },
          {
            "backgroundColor": "#00af98",
            "gap": 10,
          },
        ]
      }
      testID="magic-toast"
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight={25}
        bbWidth={25}
        fill="white"
        focusable={false}
        height={25}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "flex": 0,
              "height": 25,
              "width": 25,
            },
          ]
        }
        vbHeight={256}
        vbWidth={256}
        width={25}
      >
        <RNSVGGroup
          fill={4294967295}
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="M128 24a104 104 0 1 0 104 104A104.11 104.11 0 0 0 128 24Zm45.66 85.66-56 56a8 8 0 0 1-11.32 0l-24-24a8 8 0 0 1 11.32-11.32L112 148.69l50.34-50.35a8 8 0 0 1 11.32 11.32Z"
          />
        </RNSVGGroup>
      </RNSVGSvgView>
      <Text
        style={
          [
            {
              "color": "white",
              "fontWeight": "bold",
            },
            undefined,
          ]
        }
      >
        Taveira
      </Text>
    </View>
  </View>
</View>
`;
